cdap-ui/app/cdap/components/EntityListView/index.js

```diff
@@ -203,34 +198,55 @@ class EntityListView extends Component {
     });
   }
 
+  // Performs calculations to determine number of entities to render per page
   calculatePageSize() {
+    // different screen sizes
+    // consistent with media queries in style sheet
+    const sevenColumnWidth = 1701;
+    const sixColumnWidth = 1601;
+    const fiveColumnWidth = 1201;
+    const fourColumnWidth = 993;
+    const threeColumnWidth = 768;
+
+    // 140 = cardHeight (128) + (5 x 2 top bottom margins) + (1 x 2 border widths)
+    const cardHeightWithMarginAndBorder = 140;
+
     let entityListViewEle = document.getElementsByClassName('entity-list-view');
 
     if (!entityListViewEle.length) {
       return;
     }
-    // Performs calculations to determine number of entities to render per page
-    // minus 60px of padding in entity-list-view (30px each side)
-    let containerWidth = entityListViewEle[0].offsetWidth - 60;
 
-    // Subtract 55px to account for entity-list-header's height (30px) and container's top margin (25px)
-    // minus 20px of padding from top and bottom (10px each)
-    let containerHeight = document.getElementsByClassName('entity-list-view')[0].offsetHeight - 55 - 20;
+    // Subtract 65px to account for entity-list-info's height (45px) and paddings (20px)
+    // minus 10px of padding from top and bottom (10px each)
+    let containerHeight = entityListViewEle[0].offsetHeight - 65 - 10;
+    let containerWidth = entityListViewEle[0].offsetWidth;
+    let numColumns = 1;
+
+    // different screen sizes
+    // consistent with media queries in style sheet
+    if (containerWidth >= sevenColumnWidth) {
+      numColumns = 7;
+    } else if (containerWidth >= sixColumnWidth && containerWidth <= sevenColumnWidth-1) {
+      numColumns = 6;
+    } else if (containerWidth >= fiveColumnWidth && containerWidth <= sixColumnWidth-1) {
+      numColumns = 5;
+    } else if (containerWidth >= fourColumnWidth && containerWidth <= fiveColumnWidth-1) {
+      numColumns = 4;
+    } else if (containerWidth >= threeColumnWidth && containerWidth <= fourColumnWidth-1) {
```

Comment:
Lol d'oh... I was following the `min-width` and `max-width` media queries too closely. Updated.